Added a few stuff

wallace11 opened the pull request:
- Material Design for options & popup

- Reverted to old good css (from scss)

- Much more useful popup, with the ability to paste in links and view download history

- Add support for unrestricting folders (tested against uploaded.net) 

- Upgraded notifications including:

   1. Progress notification
   2. More informative error messages
   3. Copying unrestricted link 

-  Fixing redirection to RD torrents page after adding a magnet

- jQuery 3.1

- Commented getTraffic for now...


Apologies for the initial bunch of commits, not used to working on Github...

JDevlieghere commented on src/js/background.js:
```diff
@@ -176,33 +180,59 @@ function RealDebrid(warningPercentage, warningDays, splittingSize, torrentHost)
             if (url.match(regex)) {
                 that.urlHandler(url);
+            } else {
+                invalid = 1;
```
Why not `invalid = true`?

wallace11 replied:
You're right. I'll fix that asap!

JDevlieghere commented on src/js/background.js:
```diff
@@ -348,15 +414,25 @@ function Notifier() {
+            options.buttons = [{ title: "Copy URL To Clipboard", iconUrl: "/icons/copy_18dp_2x.png" }];
+        }
+        if (progress > 0) {
+            chrome.notifications.update("prog_" + id, options, function(){
```
Make "prog_" an ALL_CAPS constant

wallace11 replied:
Not exactly sure what you mean by that. Should we convert all the prefixes for notifications to caps?

JDevlieghere replied:
I just meant that you could make the prefixes constants like `var NOTIFICATION_ID_PREFIX = 'id_'`. It's less error-prone and more maintainable.

JDevlieghere commented:
Hi Wallace,

Thank you for the Pull-Request! I quickly glanced over the code at it seems like you made a lot of improvements, which is of course greatly appreciated. I added a few minor comments on things that caught my eye but nothing that would prevent me from merging it in. I wonder why you got rid of the SCSS though? 

I'll test drive it for a day and have @stefanpante look it over before I pull in your changes.

Thanks again for the contribution! 

Cheers,
Jonas

wallace11 commented:
Hi Jonas,

Thank you very much for the quick reply. I appreciate any comments on the changes I've made. 
There are a few places in which I felt that a different method should be used (including in my own changes), and I added a "TODO" comment in some of them. 

As for the SCSS > CSS transition - I used Material Design Lite (MDL) since I figured it was much lighter and suited for an extension than other MD implementations such as Polymer (which, by the way, uses a huge amount of memory). I thought of MDL as a complete MD solution, but later found it out to be a rather incomplete and inconsistent one. A V2 is apparently on its way (maybe with the release of Android N?), and when it's out I plan to review the design part again. Anyhow,  it took me a while to tame this beast, and working directly with CSS was much easier. 
Are there any benefits for using SCSS over CSS in such a small project?

BTW, I edited the third point on my pull-request, seems like half a sentence was missing there ;)

Cheers,
Wallace

wallace11 commented:
Alright, so I implemented your suggestions, fixed a very embarrassing miss on my end, where I didn't revert changes I made for testing, and did some more cleanup. 
Should I create a new pull request for the new commit...?

JDevlieghere commented:
Awesome! You should be able to just push it to your repository and the PR should be updated automatically. (More info on [SO](http://stackoverflow.com/questions/9790448/how-to-update-a-pull-request))

stefanpante commented on src/js/background.js:
```diff
@@ -176,33 +180,59 @@ function RealDebrid(warningPercentage, warningDays, splittingSize, torrentHost)
                 } else {
-                    nf.basic(result.message, url);
+                    nf.error("Error adding magnet");
                 }
```
for the sake of consistency, every string should be declared the same way.  with single quotes or double quotes. not both

wallace11 replied:
I just checked and all notifications use double quotes. 
Reverting all to single quotes means that any word like "doesn't" will become a little bit of a pain, wouldn't it?

stefanpante replied:
all strings means also your urls, etc. Maybe we should add an .eslintrc file in the future to enforce style.

wallace11 replied:
Gotcha. Will convert all to double quotes.
